Assets/Scripts/Scenes/InGame/Enemy/EnemySearch.cs

```diff
@@ -35,34 +42,130 @@ void Start()
-                    _myVisivilityMap.CheckVisivility(this.transform.position, _visivilityRange);
+                    //�����Ȃ��̂𒲂ׂ�B����͌���I�Ȃ��̂قǗD�悵�ĔF������
+                    if (false)
+                    { //�v���C���[�̎p�������邩���ׂ�
+
+                    }
+                    else if (false)
+                    { //���������邩���ׂ�
 
+                    }
+                    else if (_enemyStatus.ReturnEnemyState == EnemyState.Patorolling || _enemyStatus.ReturnEnemyState == EnemyState.Searching)//�v���C���[�����𗧂ĂĂ��邩�ǂ���
+                    {
+                        if (_playerStatus.nowPlayerActionState == PlayerActionState.Sneak)
+                        {
+                            if (Mathf.Pow((float)(_playerStatus.nowPlayerSneakVolume * _audiomaterPower * 0.01f), 2f) - (Mathf.Pow(transform.position.x - _player.transform.position.x, 2) + (Mathf.Pow(transform.position.y - _player.transform.position.y, 2))) > 0)
+                            { //�����������邩�ǂ���
+                                if (_debugMode) Debug.Log("�E�ԉ�����������");
+                                _myVisivilityMap.HearingSound(_player.transform.position,  15);
+                                _myEneyMove.SetMovePosition(_myVisivilityMap.GetNextNearWatchPosition(this.transform.position));
+                            }
+                        }
+                        else if (_playerStatus.nowPlayerActionState == PlayerActionState.Walk)
+                        {
+                            if (Mathf.Pow((float)(_playerStatus.nowPlayerWalkVolume * _audiomaterPower * 0.01f), 2f) - (Mathf.Pow(transform.position.x - _player.transform.position.x, 2) + (Mathf.Pow(transform.position.y - _player.transform.position.y, 2))) > 0)
+                            { //�����������邩�ǂ���
+                                if (_debugMode) Debug.Log("����������������");
+                                _myVisivilityMap.HearingSound(_player.transform.position,  15);
+                                _myEneyMove.SetMovePosition(_myVisivilityMap.GetNextNearWatchPosition(this.transform.position));
+                            }
+                        }
+                        else if (_playerStatus.nowPlayerActionState == PlayerActionState.Dash)
+                        {
+                            if (Mathf.Pow((float)(_playerStatus.nowPlayerRunVolume * _audiomaterPower * 0.01f), 2f) - (Mathf.Pow(transform.position.x - _player.transform.position.x, 2) + (Mathf.Pow(transform.position.y - _player.transform.position.y, 2))) > 0)
+                            { //�����������邩�ǂ���
+                                if (_debugMode) Debug.Log("���鉹����������");
+                                _myVisivilityMap.HearingSound(_player.transform.position, 15);
+                                _myEneyMove.SetMovePosition(_myVisivilityMap.GetNextNearWatchPosition(this.transform.position));
+                            }
+                        }

```

Comment:
```suggestion
float valume = 0;
switch(_playerStatus.nowPlayerActionState){
    case PlayerActionState.Sneak:
        valume = _playerStatus.nowPlayerSneakVolume;
        break;
    case PlayerActionState.Walk:
        valume = _playerStatus.nowPlayerWalkVolume;
        break;
    case PlayerActionState.Dash:
        valume = _playerStatus.nowPlayerRunVolume;
        break;
}
if (Mathf.Pow((float)(valume* _audiomaterPower * 0.01f),2f)
    - (Mathf.Pow(transform.position.x - _player.transform.position.x,2) 
    + (Mathf.Pow(transform.position.y - _player.transform.position.y, 2))) > 0) 
{ //�����������邩�ǂ���
    _myVisivilityMap.HearingSound(_player.transform.position, 15);
    _myEneyMove.SetMovePosition(_myVisivilityMap.GetNextNearWatchPosition(this.transform.position));
}
``` 
上に書いたみたいにこんな感じでまとめられませんか